src/confopt/searchspace/common/mixop.py

```diff
@@ -303,3 +327,24 @@ def forward(self, x: torch.Tensor) -> torch.Tensor:
         out = torch.cat([self.conv_1(x), self.conv_2(x[:, :, 1:, 1:])], dim=1)
         out = self.bn(out)
         return out
+
+
+class DynamicAttentionNetwork(nn.Module):
+    def __init__(self, C: int, num_ops: int, attention_weight: float) -> None:
+        super().__init__()
+        self.fc1 = nn.Linear(C, C // 4)
+        self.fc2 = nn.Linear(C // 4, num_ops)
+        self._attention_weight = attention_weight
+
+    def update_attention_weight(self, attention_weight: float) -> None:
+        self._attention_weight = attention_weight
+
+    def forward(self, x: torch.Tensor) -> torch.Tensor:
+        out = F.adaptive_avg_pool2d(x, output_size=(1, 1))
+        out = out.view(x.size(0), -1)  # shape (N, C)
+
+        out = F.relu(self.fc1(out))
+        out = F.relu(self.fc2(out))
+
+        out = out.mean(dim=0)
```

Comment:
This is a a bit confusing for me about how to handle batches. The paper never mention how do handle batches. So i averaged over them, but I cannot verify if this is a correct thing to do.

![image](https://github.com/user-attachments/assets/40d06157-9f43-4454-bbf4-fbcdb03b41dd)

<img width="719" alt="image" src="https://github.com/user-attachments/assets/8be236d3-dcbd-4b54-932f-7b27e99c255c" />

Reply:
The authors are indeed vague about how they handle the batch dimension, and there is no code available. We can stick to this implementation.